src/container.c

```diff
@@ -789,15 +789,52 @@ struct hyper_container *hyper_find_container(struct hyper_pod *pod, const char *
 	return NULL;
 }
 
-void hyper_cleanup_container(struct hyper_container *c, struct hyper_pod *pod)
+static void hyper_cleanup_container_mounts(struct hyper_container *container, struct hyper_pod *pod)
 {
-	char root[512];
+	int pid, pipe[2] = {-1, -1};
 
-	sprintf(root, "/tmp/hyper/%s/devpts/", c->id);
-	if (umount(root) < 0 && umount2(root, MNT_DETACH))
-		perror("umount devpts failed");
+	if (pipe2(pipe, O_CLOEXEC) < 0) {
+		perror("create pipe for unmount failed");
+		return;
+	}
+
+	pid = fork();
+	if (pid < 0) {
+		perror("fork unmount process failed");
+		goto out;
+	} else if (pid == 0) {
+		if (hyper_enter_sandbox(pod, -1) < 0) {
```

Comment:
the second parameter of hyper_enter_sandbox is -1 here, it seems like we don't need pipe, waitpid is enough, right?